spinnaker_components/common/fixed_point.h

```diff
@@ -0,0 +1,97 @@
+/* Optimised fixed point methods.
+ */
+
+#include "arm_acle_gcc_selected.h"
+#include "nengo_typedefs.h"
+
+#ifndef __FIXED_POINT_H__
+#define __FIXED_POINT_H__
+
+/*****************************************************************************/
+// Convert from an INT64 to appropriate INT32 fixed point representation
+
+static inline int32_t scale_64_to_32(int64_t value)
```

Comment:
This is a bit ambiguous  - convert_s32_30_s16_15 is also gross, but maybe more descriptive..